build/Linux+BSD/portable/RecipeDebian

```diff
@@ -0,0 +1,176 @@
+#!/bin/bash
+
+# This file is designed to run in debian jessie docker images.
+
+COMMAND=$1
+ARCH=$2
+
+main() {
+
+  set -e # Halt on error
+  set -x # Be verbose
+
+  shift 2 #need to adjust parameter list so $makefile_overrides start at $1
+
+  case $COMMAND in
+    "--fetch-build-dependencies-only")
+      fetch-build-dependencies
+      ;;
+    "--fetch-package-dependencies-only")
+      fetch-package-dependencies
+      ;;
+    "--build-only")
+      build "$@"
+      ;;
+    "--package-only")
+      package
+      ;;
+    *)
+      fetch-build-dependencies
+      fetch-package-dependencies
+      build "$@"
+      package
+  esac
+
+  echo 'RecipeDebianCross completed'
+}
+
+#################################################################################
+# Fetch compile-time dependencies only
+##################################################################################
+fetch-build-dependencies() {
+  echo "fetching dependencies for arm arch: $ARCH"
+
+  # get cross compiler toolchain
+  apt-key adv --fetch-keys http://emdebian.org/tools/debian/emdebian-toolchain-archive.key
+  echo "deb http://emdebian.org/tools/debian/ jessie main" > /etc/apt/sources.list.d/crosstools.list
+
+  # allow to download $ARCH-specific packages
+  dpkg --add-architecture $ARCH
+
+  apt-get clean && apt-get update && apt-get install -y \
+    libsndfile1-dev:$ARCH \
+    libasound2-dev:$ARCH \
+    portaudio19-dev:$ARCH \
+    zlib1g-dev:$ARCH \
+    libfreetype6-dev:$ARCH \
+    libmp3lame-dev:$ARCH \
+    libssl-dev:$ARCH \
+    libdrm-dev:$ARCH \
+    libgl1-mesa-dev:$ARCH \
+    libpulse-dev:$ARCH \
+    qtbase5-dev:$ARCH \
+    qttools5-dev:$ARCH \
+    qtscript5-dev:$ARCH \
+    libqt5xmlpatterns5-dev:$ARCH \
+    libqt5svg5-dev:$ARCH \
+    libqt5webkit5-dev:$ARCH \
+    qttools5-dev-tools:$ARCH \
+    cross-gcc-dev \
+    binutils-arm-linux-gnueabihf \
+    crossbuild-essential-armhf \
+    qemu-user-static \
+    cmake \
+    git
+
+  # forcibly install because will break aptitude dependency rules 
+  force-install qtbase5-dev-tools $ARCH
+}
+
+###################################################################################
+# Fetch run-time dependencies only...to be run after fetching compile dependencies
+###################################################################################
+fetch-package-dependencies() {
+  echo "fetching AppImage packaging + runtime dependencies for arch: $ARCH"
+
+  force-install \
+    alsa-base any \
+    alsa-tools $ARCH \
+    alsa-utils $ARCH \
+    libasound2-plugins $ARCH \
+    pulseaudio $ARCH \
+    pkg-config $ARCH \
+    liblcms2-2 $ARCH \
+    libmng1 $ARCH \
+    libicu52 $ARCH \
```

Comment:
@shoogle I've finally figured out what is going on with this...it has something to do with using .so instead of .so.52, which we decided to do becuase the CentOS 6 names these .so.42, while Jessie names them .so.52.  One simply way around this issue seems to be force removal of libicu52:amd64 using "dpkg --purge --force-all", and then somehow it copies the correct one...  But I'd like something better.  If I can't think of anything better, I'm putting these 3 libs in the OS if statement...